src/main/java/org/devnull/jedi/PowerDNSConnectionHandler.java

```diff
@@ -74,6 +74,7 @@ public PowerDNSConnectionHandler(Socket client,
 		this.apiPool = apiPool;
 		this.cache = cache;
 		restClient = new RestClient(config);
+		socket.setSoTimeout(config.unix_socket_timeout);
```

Comment:
You're configuring every socket with the *unix* socket timeout here.  Shouldn't that be done in the UnixSocketServer class in Jedi instead?  The PowerDNSConnectionHandler handles **all** sockets.